### PR TITLE
chore(file): Removes the class FilePluginFile

### LINKED ISSUE #7763: Remove FilePluginFile
We just use this to tack on a custom delete method, and this only works when we use the FilePluginFile constructor. This confuses the entity persistence model and will confuse users. See also [these comments](https://github.com/mrclay/Elgg-leaf/blob/7761_filepluginfile/mod/file/classes/FilePluginFile.php).

### PR DESCRIPTION
**Pull request by mrclay**
BREAKING CHANGE:
If you use the class FilePluginFile in your plugin, replace this usage with ElggFile (for construction). Load files objects with get_entity().

Fixes #7763

### CODE REVIEW

File: docs/guides/upgrading.rst [hunk 1]
```diff
@@ -133,14 +135,19 @@ Plugin Messages
...
+
 Removed Functions
 -----------------
 
- - get_db_error()
- - execute_delayed_query()
- - get_db_link()
- - load_plugins()
- - mysql_*()
+ - ``get_db_error()``
```
Comment (hypeJunction) [inline docs/guides/upgrading.rst]: Were these meant as a separate commit?

**Comment by hypeJunction** [comment]
Aside from the noise in the docs, LGTM

File: docs/guides/upgrading.rst (hunk 1 above)
Comment (mrclay) [inline docs/guides/upgrading.rst]: Eh I got lazy, I'll move them out

Comment (hypeJunction) [inline docs/guides/upgrading.rst]: It's fine by me, unless someone else objects

**Comment by mrclay** [comment]
Cleaned up docs. Can we get a second :+1:?

File: mod/file/classes/FilePluginFile.php [hunk 2]
```diff
@@ -1,42 +1,15 @@
 <?php
 
 /**
- * Override the ElggFile
- *
- * @note When extending an ElggEntity, one should always register a unique type/subtype
- *       combination. We failed to do so here, so get_entity() and friends will always return
- *       these objects as ElggFile. We leave it this way just for BC.
+ * Removed subclass of ElggFile. See https://github.com/Elgg/Elgg/issues/7763
```
Comment (juho-jaakkola) [inline mod/file/classes/FilePluginFile.php]: Add a mention that this was removed in 2.0 so we'll remember to remove it for 3.0.

**Comment by juho-jaakkola** [comment]
LGTM

**Pull request merged**